src/components/overlays/badge/Badge.vue

```diff
@@ -39,11 +40,11 @@ const slots = useSlots();
 
 const { modelValue, offsetX, offsetY, icon, text } = toRefs(props);
 
-const hasIcon = computed(() => !!get(icon) || !!slots.icon);
```

Comment:
I did a small change here because the explicit `!!` is not needed with the logicOr function and also doing `!!get(x)` breaks reactivity since we unref icon in an non reactive context.